packages/ng/src/app/select/select.directive.ts

```diff
@@ -0,0 +1,257 @@
+import {
+	OnChanges,
+	Directive,
+	OnInit,
+	Input,
+	Output,
+	EventEmitter,
+	forwardRef,
+	OnDestroy,
+	Renderer2,
+	ElementRef,
+	SimpleChanges,
+	ViewContainerRef,
+	HostListener,
+	HostBinding
+} from '@angular/core';
+import {
+	Overlay,
+	OverlayConfig,
+} from '@angular/cdk/overlay';
+import {
+	NgModel,
+	ControlValueAccessor,
+	NG_VALUE_ACCESSOR,
+	NG_VALIDATORS,
+	Validator,
+	ValidatorFn,
+	ValidationErrors,
+	AbstractControl
+} from '@angular/forms';
+import { Subscription } from 'rxjs/Subscription';
+
+import { LuPopoverTrigger } from '../popover';
+import { LuSelectPopover } from './select.popover.component';
+import { LuSelectOption } from './select.option.component';
+
+/** KeyCode for End Key */
+const END = 'End';
+/** KeyCode for Home Key */
+const HOME = 'Home';
+/** KeyCode for Escape Key */
+const ESCAPE = 'Escape';
+/** KeyCode for Backspace Key */
+const BACKSPACE = 'Backspace';
+/** KeyCode for Delete Key */
+const DELETE = 'Delete';
+/** KeyCode for Enter Key */
+const ENTER_KEY = 'Enter';
+/** KeyCode for ArrowUp Key */
+const UP_KEY = 'ArrowUp';
+/** KeyCode for ArrowDown Key */
+const DOWN_KEY = 'ArrowDown';
+
+/**
+ * Directive to put on a div to allow it to react with a popover to emulate a select component
+ */
+@Directive({
+	selector: 'div[luSelect]',
```

Comment:
it was a long discussion with @jeremie-lucca about the use or not of an input and as the possibility to search in a select will be include the popover, it makes no sense to have an input as directive selector. Indeed, in a HTML Select, you will never enter text in the area that looks likes the input. So as there will be no text enter in this area, we don't need to have an input.

luApiPicker is more like an autocomplete rather than a select

Reply:
so then why not `select[luSelect]`
i find it very ugly to have a `ngModel` on a div

Reply:
or just `[luSelect]`

Reply:
Using a div rather than an input inside the "textfield" template breaks it in mod-framed @jeremie-lucca . I second Lucien for the idea of using an `input` or a `select` for accessibility sakes. A div can't be focused through keyboard interactions.

Reply:
just [luSelect] is to open in term of selector, so we can use it with to many elements that won't respond well. select[luSelect] isn't either what we want due to lack of customization. 

And input for something that is not an input isn't a good choice to.

Reply:
@sanlucca the div is focusable, i update it dynamicly with the directive.

If you look at shadow dom of an html select, you will see that it is not input 

![image](https://user-images.githubusercontent.com/681267/35052426-d4da3d90-fba7-11e7-83bb-4d6d38d48233.png)
and if you look at what [select-2 does](https://select2.org/getting-started/basic-usage), they don't have an input to but a span : 
![image](https://user-images.githubusercontent.com/681267/35052536-1b23440e-fba8-11e7-8c15-11ffcb8c1839.png)

and finally same thing for the [select of angular marterial](https://material.angular.io/components/select/examples) : 
![image](https://user-images.githubusercontent.com/681267/35052645-632eaf22-fba8-11e7-9d1b-25f3e84dd2bf.png)

Reply:
Then we should add `role="listbox"` attribute to the component and `role="option"` to its options.
[More info there](https://developer.mozilla.org/fr/docs/Accessibilit%C3%A9/ARIA/Techniques_ARIA/Utiliser_le_role_listbox)

Reply:
we're gonna enter spome philosophic debate fyi

> it makes no sense to have an input as directive selector
i disagree, it depends on what role the tag input has for you. For me it is the tag to use if you want a value to be editable by the end user. with this definition it _should_ be on an input tag.

moreover, when you say

> because they wont type stuff in it to filter the number of options, it shouldnt be an input

does that means that the next component `luAutocomplete` willhave this selector `input[luAutocomplete]`, then if i want to change all my selects to autocomplete cuz i want all dropdowns in my app to behave the same, then i also have to change div to input

also you cant type stuff in `<input type="checkbox"> does thjat mean it shouldnt be a tag input ?

once again for me you mix up between what it is and how it behaves.
 - an `input` is a tag to bind a value to the view and make it editable, some sub tags can be `select` or `textarea`
 - a div is a tag to display sutff

in this particular case what is inside the input, the `elementRef.nativeElement.value`, isn't directly editable, or in your future behaviours wont be use to filter the options cuz you'll add a header in the picker popover, so its not an input

-----------

if you look at the formly overlay, you can see that each type template looks exactly the same
```html
<input of some kind>
<label>the label</label
```
it is coherent cuz people understand that an input is where the value will be binded in edit mode and not jus in display mode.

using a div here is really baffling to me

Reply:
In a near furture, there will be the possibility to search in the Popover, which is still different as an autocomplete. An autocomplete for me should return or map a string. 

So Autocomplete is different in term of use as a select for me. So if you should transform an auto complete to select or the inverse, it's something weird. But i think it's a point of view of UX.

About the `div`, I think that could be a `span` but, event if formly do inputs, Other vendors that do selects, don't uses inputs.

Reply:
i think they use spans for technical reasons, but my point still stands.

in my head its an input before being anything else : its a way for the end user to _input_ a value. 

if you did a new component for tackling ribs or email adresses, would you put it on a `input` or a `div` ?

at least try `select[luselect]`, see if it works

Reply:
I have to reopen this comment because i was trying to refactor my code to implement the solution with an `input` but I face somes problems that leads me to the conclusion that, using an input isn't maybe the best solution.....

If you use `input`, indeed it's better in term of  concepts of manipulating datas but you have to override the whole behaviour of an `input`. Let me explain myself. As the text area in a `select` isn't editable, you have to things like that : 
* avoid all text typing => add a prevent.default for all Keys except the ones tolerate by the select (Up, Down, ....)
* Override the focus => remove the selection when the element is focused
* Apply a hack in the style to hide the text selection cursor => change the color to transparant and apply a text-shadow corresponding to the real color of the text. 

I cannot add dynamicly disabled or readonly to the `input` because, How the developer could specify that he want to disabled the field ? Maybe we could play with css `class` but it will force the developer to add an additionnal stylesheet when he will use this component.

An other solution could be to dynamcly hide the original input and replace it by something else but again, when the developer want to style it, it will be very disturbing

So Here are the potential solutions with pros and cons : 

**Use an input**
_Pros_ : 
* In term of dom, it's better for the developper because as you say, an input is something that we use to manipulate datas
_Cons_ : 
* You have to override the whole behaviour of a real input, and it sounds more like a hack THAN like something natural...


**Use an input but replace it dynamicly**
_Pros_ :
* In term of dom, it's better for the developper because as you say, an input is something that we use to manipulate datas
_Cons_:
* When the dev want to target the field, it could be disturbing because, he don't have to target the `input` but something else...
* It could impact the tree dom styling hierarchy

**Use a div, span, or something else that is not an input**
_Pros_: 
* Except the focus management, you won't have to override anything else and it could offer you more possibilities to add dynamic content like "clear"
_Cons_:
* It's strange for the developer to use a div when he want to manipulate datas...

So @sanlucca, @lucienbertin, @jeremie-lucca, what did you prefer ?

The three solutions are possible but I prefer maybe the last solution I don't have the feeling to do a hudge hack just to have a meaning full dom tree.

Please take the time to thing about other pros / cons

Reply:
I'd rather go to the div/span solution to keep the DOM as simple as possible. 
But if input is a technical requirement, having a hidden input / select or select with "multiple" attribute on it, is ok for me.